src/core/ddsi/src/ddsi_cdrstream.c

```diff
@@ -1553,6 +1459,35 @@ static const uint32_t *dds_stream_read_delimited (dds_istream_t * __restrict is,
   return ops;
 }
 
+static bool dds_stream_read_pl_member (dds_istream_t * __restrict is, char * __restrict data, uint32_t m_id, const uint32_t * __restrict ops)
+{
+  uint32_t insn, ops_csr = 0;
+  bool found = false;
+
+  /* FIXME: continue finding the member in the ops member list starting from the last
+      found one, because in many cases the members will be in the data sequentially */
+  while (!found && (insn = ops[ops_csr]) != DDS_OP_RTS)
+  {
+    assert (DDS_OP (insn) == DDS_OP_PLM);
+    uint32_t flags = DDS_PLM_FLAGS (insn);
+    const uint32_t *plm_ops = ops + ops_csr + DDS_OP_ADR_PLM (insn);
+    if (flags & DDS_OP_FLAG_BASE)
+    {
+      assert (DDS_OP (plm_ops[0]) == DDS_OP_PLC);
+      plm_ops++; /* skip PLC to go to first PLM from base type */
```

Comment:
Why not take care of this off-by-1 in the generator?

Reply:
Again readability of the generated opcodes, because all offsets to a type are to the PLC/DLC instruction of the target type (when it's not a final type of course), and I preferred to keep it that way to make debugging easier. The same applies to appendable struct inheritance: the ADR | EXT could also point to the first instruction after the DLC. But on the other hand this will slightly impact runtime performance, so I'm also fine with handling this in the generator (in which case I could include a comment in the generated opcodes).

Reply:
I'm not worried about the run-time impact of the additional increment: with all the work needed to decode the operations and make sense of the data/generating the required format, the overhead really is mere noise.

I haven't been doing much debugging of this code, unlike you, so I'm not in a good position to judge the effect on the difficulty debugging it. I'm quite willing to accept your reasoning.